src/components/MintDialog/pages/NativeMint/NativeMint.tsx

```diff
@@ -42,15 +43,33 @@ export const NativeMint: FC<NativeMintProps> = ({
   const network = useNetwork()
 
   const wrongChain = network.chain?.id !== l2.id
-  const { info: {creatorAddress, dropName, crossMintClientId, mintDotFunStatus} } =
-    useMintDialogContext()
+  const {
+    info: {
+      creatorAddress,
+      dropName,
+      crossMintClientId,
+      mintDotFunStatus,
+      dropDataSuffix,
+    },
+    setInfo,
+  } = useMintDialogContext()
   const isPendingConfirmation =
     page === ModalPage.NATIVE_MINT_PENDING_CONFIRMATION
   const isPendingTx = page === ModalPage.NATIVE_MINTING_PENDING_TX
   const isPending = isPendingConfirmation || isPendingTx
 
   const isMintDotFun = typeof mintDotFunStatus === 'object'
 
+  const handleDataSuffixChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setInfo((prevState) => ({
+        ...prevState,
+        dataSuffix: e.target.checked ? dropDataSuffix!.value : siteDataSuffix,
```

Comment:
Should this be like this, since we're already passing in dataSuffix here and set value to siteDataSuffix if not present? 
```suggestion
        dataSuffix: e.target.checked ? dropDataSuffix!.value : dataSuffix,
```

Reply:
The way you are suggesting, once the checkbox is checked users won't be able to undo their selection. Because dataSuffix would have changed to dropDataSuffix. So it will be as if the user checked.